modules/src/core/ics04_channel/context.rs

```diff
@@ -138,6 +138,48 @@ pub trait ChannelReader {
     }
 }
 
+/// Defines a subset of the `ChannelReader`'s methods that a light-client implementation can access.
+/// A blanket implementation of this trait is provided for all types that implement `ChannelReader`.
+pub trait ChannelMetaReader {
+    /// Returns the current height of the local chain.
+    fn host_height(&self) -> Height;
+
+    /// Returns the current timestamp of the local chain.
+    fn host_timestamp(&self) -> Timestamp;
+
+    /// Returns the time when the client state for the given [`ClientId`] was updated with a header for the given [`Height`]
+    fn client_update_time(&self, client_id: &ClientId, height: Height) -> Result<Timestamp, Error>;
+
+    /// Returns the height when the client state for the given [`ClientId`] was updated with a header for the given [`Height`]
+    fn client_update_height(&self, client_id: &ClientId, height: Height) -> Result<Height, Error>;
+
+    /// Calculates the block delay period using the connection's delay period and the maximum
+    /// expected time per block.
+    fn block_delay(&self, delay_period_time: Duration) -> u64;
+}
+
+impl<T: ChannelReader> ChannelMetaReader for T {
```

Comment:
This is definitely something we should do once we remove the `Any*` enums. I didn't do it in this PR because some of the trait methods have default impls that depend on other trait methods that use the `Any*` enums. (e.g. `host_timestamp()` and `pending_host_consensus_state()`)